src/parser.yy

```diff
@@ -314,10 +314,17 @@ expr : int                                      { $$ = $1; }
      | MINUS expr                               { $$ = new ast::Unop(token::MINUS, $2, @1); }
      | MUL  expr %prec DEREF                    { $$ = new ast::Unop(token::MUL,  $2, @1); }
      | expr DOT ident                           { $$ = new ast::FieldAccess($1, $3, @2); }
+     | expr DOT INT                             { $$ = new ast::FieldAccess($1, $3, @3); }
```

Comment:
For the parser  I think  `map_or_var DOT INT` makes more sense here.  The downside is the error it generates

```
stdin:1:9-16: ERROR: syntax error, unexpected integer
i:s:1 { (1+2).2 }
        ~~~~~~~
```

Although I don't like the current one that much either:

```
stdin:1:9-16: ERROR: Can not access field '' on expression of type 'int64'
i:s:1 { (1+2).2 }
        ~~~~~~~
```

Reply:
I think this would bread nested tuple indexing:
```
[  FAILED  ] basic.basic tuple
        Command: ../src/bpftrace -e 'BEGIN { $v = 99; $t = (0, 1, "str", (5, 6), $v); printf("%d %d %s %d %d %d\n", $t.0, $t.1, $t.2, $t.3.0, $t.3.1, $t.4); exit(); }'
        Expected: 0 1 str 5 6 99
        Found: stdin:1:98-104: ERROR: syntax error, unexpected integer
BEGIN { $v = 99; $t = (0, 1, "str", (5, 6), $v); printf("%d %d %s %d %d %d\n", $t.0, $t.1, $t.2, $t.3.0, $t.3.1, $t.4); exit(); }
```

Reply:
I think I can make the error message better too